### PR TITLE
Update types used for image creation in Dockerode

### PR DESCRIPTION
**Pull request by thegecko**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
All file-based functions also accept streams: https://github.com/apocas/dockerode/blob/master/lib/docker.js#L184
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

_Notes_

- Changes were found/tested during the usage of Dockerode with TypeScript
- The return type for createImage is actually `HTTP.IncomingMessage`, but followed the style of using `any` in the same way as similar methods
- IMO, `any` return types should be replaced with correct types in a future PR

### CODE REVIEW

File: types/dockerode/index.d.ts [hunk 1]
```diff
@@ -832,18 +833,18 @@ declare class Dockerode {
   createContainer(options: Dockerode.ContainerCreateOptions, callback: Callback<Dockerode.Container>): void;
   createContainer(options: Dockerode.ContainerCreateOptions): Promise<Dockerode.Container>;
 
-  createImage(options: {}, callback: Callback<Dockerode.Image>): void;
-  createImage(auth: any, options: {}, callback: Callback<Dockerode.Image>): void;
-  createImage(options: {}): Promise<Dockerode.Image>;
-  createImage(auth: any, options: {}): Promise<Dockerode.Image>;
+  createImage(options: {}, callback: Callback<any>): void;
```
Comment (Seikho) [inline types/dockerode/index.d.ts]: Why are you changing these `any`?

**Review by Seikho — CHANGES_REQUESTED** [review]

File: types/dockerode/index.d.ts (hunk 1 above)
Comment (Seikho) [reply types/dockerode/index.d.ts]: After digging into this, it looks like it returns a `NodeJS.ReadableStream`

Comment (thegecko) [reply types/dockerode/index.d.ts]: As I outlined in my PR message, it actually returns an 'HTTP.IncomingMessage', but I changed it to 'any' to be consistent with all the other methods which also just return 'any'. I prefer consistency when changing code and believe all of the 'any' return types should be aligned for now and then fixed together later. This is out of scope for this PR.

Comment (Seikho) [reply types/dockerode/index.d.ts]: The API surface is pretty large in `dockerode` so I prefer to just chip away at them as I use that part of the API. I originally submitted these types, I simply didn't have the time or patience to correctly define everything.
In this case, I just got it wrong and we know what the correct type is. Luckily here it's just a `node` type and not a `Docker Remote API` type, so it's pretty easy to change it.

**Review by Seikho — APPROVED** [review]

File: types/dockerode/index.d.ts (hunk 1 above)
Comment (thegecko) [reply types/dockerode/index.d.ts]: I've added these return types for this area only. I am assuming no other type is ever returned from these methods as I've only tested it using promises and reading as a stream.

**Review by meisenzahl — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**